ext/st25r3911b/st25r3911b.h

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#ifndef ST25R3911B_H
+#define ST25R3911B_H
+
+#include <zephyr/types.h>
+#include "st25r3911b_spi.h"
+
+/**
+ * @file
+ * @defgroup st25r3911b_common ST25R3911B NFC Reader common functionality
+ * @{
+ *
+ * @brief API for the ST25R3911B NFC Reader common functionality.
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** ST25R3911B Max Tx packet length. */
+#define ST25R3911B_MAX_TX_LEN 8191
+
+/** ST25R3911B No RF Collision Avoidance */
+#define ST25R3911B_NO_THRESHOLD_ANTICOLISION 0xFF
+
+/** ST25R3911B Max RF Collision Avoidance Threshold */
+#define ST25R3911B_MAX_THRESHOLD 0x0F
+
+/** ST25R3911B FTD adjustment. */
+#define ST25R3911B_FDT_ADJUST 64
+
+/** ST25R3911B Mask Reciver timer max FTD. */
+#define ST25R3911B_MASK_RECEIVER_MAX_FDT 0xFC0
+
+/** ST25R3911B Non-Response Timer max value for step 64/fc. */
+#define ST25R3911B_NRT_64FC_MAX 0x3FFFC0
+
+/** ST25R3911B Non-Response Timer max FTD in fc. */
+#define ST25R3911B_NRT_FC_MAX 0x4095BF6A000
+
+/** Fc in 64/fc. */
+#define NFC_1FC_IN_64FC 64
+
+/** Fc in 4096/fc. */
+#define NFC_1FC_IN_4096FC 4096
+
+/**
+ * @defgroup st25r3911b_leds
+ * @{
+ * @brief ST25R3911B Leds pin.
+ */
+#define ST25R3911B_NFCF_LED 4
+#define ST25R3911B_NFCB_LED 28
+#define ST25R3911B_NFCA_LED 29
+
+/**
+ * @}
+ */
+
+/* According to ST25R3911B timer registers are set in this way:
+ * register value * 64fc or register value * 4096fc.
+ */
+
+/** Conversion FC to 64/FC. According to ST25R3911B timer
+ *  registers are set in this way:
+ *  register value * 64fc or register value * 4096fc.
+ */
+#define FC_TO_64FC(_fc) ceiling_fraction((_fc), NFC_1FC_IN_64FC)
+
+/** Conversion FC to 4096/FC. According to ST25R3911B timer
+ * registers are set in this way:
+ * register value * 64fc or register value * 4096fc.
+ */
+#define FC_TO_4096FC(_fc) ceiling_fraction((_fc), NFC_1FC_IN_4096FC)
+
+/** @brief Initialize NFC Reader
+ *
+ *  @details NFC Reader initialization common for all
+ *           NFC technology.
+ *
+ *  @return Returns 0 if initialization was successful,
+ *          otherwise negative value.
+ */
+int st25r3911b_init(void);
+
+/** @brief Set NFC Reader Tx packet length.
+ *
+ *  @details Maximum packet length is @ref ST25R3911B_MAX_TX_LEN.
+ *
+ *  @param[in] len Tx packet length.
+ *
+ *  @return Returns 0 if initialization was successful,
+ *          otherwise negative value.
+ */
+int st25r3911b_set_tx_len(u16_t len);
```

Comment:
There is no need to have a symmetric function for rx len, this function sets a reader register, because reader have to know when put the EoF and CRC, payload can be 8191 bytes long but the reader have 96 bytes build-in FIFO. In case on RX the data can be received while the end frame mark will be received so a reader do not have a register to set this value.